meetings/2017/05-18.md

```diff
@@ -19,11 +22,330 @@ Attendees
 - Would be good to have Flow in sooner rather than later.
   - If TypeScript definitions change if we merged it later, then it could possibly break (low risk).
 
+## 5.x recap
+
+Let's recap here major Breaking changes / changes in 5.x vs 4.x
+
+Please correct me if I'm wrong in any of those
+
+### Component reference via Constructor
+
+> isn't possible in 5.x ( Preact doesn't recognises custom-element via CTOR )
+> if you wanna be able to use it in 5.x you have to use [val](https://github.com/skatejs/val) which may be too much 
+
+**4.x**
+
+```js
+// MyCarousel.jsx
+import { h, Component, define } from 'skate'
+
+class MyCarousel extends Component {
+  static get is(){ return 'my-carousel' }
+}
+
+export define(MyCarousel)
+
+// MyComponent.jsx
+import { h, Component, define } from 'skate'
+import { MyCarousel } from './MyCarousel'
+ 
+class MyComponent extends Component {
+  static get is(){ return 'my-cmp' }
+  renderCallback(){
+    return (
+      <div>
+        <h1>Hello World</h1>
+        <MyCarousel images={['./path/foo.jpg']} interval={1000} />
+      </div>
+    )
+  }
+}
+
+define(MyComponent)
+```
+
+**5.x**
+
+```js
+// MyCarousel.jsx
+import { h, Component, define } from 'skate'
+class MyCarousel extends Component {
+  static get is(){ return 'my-carousel' }
+}
+
+export define(MyCarousel)
+
+// MyComponent.jsx
+import { h, Component, define } from 'skate'
+import { MyCarousel } from './MyCarousel'
+ 
+class MyComponent extends Component {
+  static get is(){ return 'my-cmp' }
+  renderCallback(){
+    return (
+      <div>
+        <h1>Hello World</h1>
+        <my-carousel images={['./path/foo.jpg']} interval={1000} />
+        {/* OR via string refference */}
+        <MyCrousel.is images={['./path/foo.jpg']} interval={1000} />
+      </div>
+    )
+  }
+}
+
+define(MyComponent)
+```
+
+### return Array as root node
+
+> isn't possible in 5.x ( Preact doesn't support rendering arrays as React Fiber for instance )
+> Critical refactoring especially for apps that are using custom-elements for layout via Flexbox
```

Comment:
Maybe it'd be best to have a compat package that can make things smoother. As mentioned in previous meetings, you can have several versions of Skate side-by-side, so I'm not sure this is a huge issue. Could also write code shifts for it.